docs/source/reference/core/dataset.rst

```diff
@@ -0,0 +1,42 @@
+.. module:: chainer.dataset
+
+Dataset abstraction
+===================
+
+Chainer has a support of common interface of training and validation datasets. The dataset support consists of three components: datasets, iterators, and batch conversion functions.
+
+**Dataset** represents a set of examples. The interface is only determined by combination with iterators you want to use on it. The built-in iterators of Chainer requires the dataset to support ``__getitem__`` and ``__len__`` method. In particular, the ``__getitem__`` method should support indexing by both an integer and a slice. We can easily support slice indexing by inheriting :class:`DatasetMixin`, in which case users only have to implement :meth:`~DatasetMixin.get_example` method for indexing. Some iterators also restrict the type of each example. Basically, datasets are considered as `stateless` objects, so that we do not need to save the dataset as a checkpoint of the training procedure.
+
+**Iterator** iterates over the dataset, and at each iteration, it yields a mini batch of examples as a list. Iterators should support the :class:`Iterator` interface, which includes the standard iterator protocol of Python. Iterators manage where to read next, which means they are `stateful`.
+
+**Batch conversion functions** convert the mini batch into arrays to feed to the neural nets. They are also responsible to send each array to an appropriate device. Chainer currently provides :func:`concat_example` as the only example of batch conversion functions.
```

Comment:
`Dataset` and `Iterator` is singular, while `Batch conversion functions` is prural.